resources/patches/scheduler_workbench-allowed_status.patch

```diff
@@ -0,0 +1,147 @@
++ */
++function scheduler_workbench_update_7001() {
++  if (!variable_set('scheduler_workbench_allowed_status', array())) {
++    _scheduler_workbench_status();
++  }
++}
++
++/**
++ * Sets the allowed status that can be scheduled for publication.
++ */
++function _scheduler_workbench_status() {
++  $transitions = workbench_moderation_transitions();
++  $status = array();
++  foreach ($transitions as $key => $value) {
++    if ($value->to_name != 'published') {
++      unset($transitions[$key]);
++    }
++    else {
++      $status[] = $value->from_name;
++    }
++  }
++  variable_set('scheduler_workbench_allowed_status', $status);
++}
+diff --git a/scheduler_workbench.module b/scheduler_workbench.module
+index 868435e..0d44b5a 100755
+--- a/scheduler_workbench.module
++++ b/scheduler_workbench.module
+@@ -7,6 +7,22 @@
+  */
+
+ /**
++ * Implements hook_menu().
++ */
++function scheduler_workbench_menu() {
++  $items['admin/config/content/scheduler_workbench'] = array(
++    'title' => 'Workbench Scheduler publication states',
++    'description' => 'Select the states that can be scheduled for publication',
++    'page callback' => 'drupal_get_form',
++    'page arguments' => array('scheduler_workbench_allowed_status_form'),
++    'access arguments' => array('administer scheduler workbench'),
++    'file' => 'scheduler_workbench.admin.inc',
++    'weight' => 0,
++    'type' => MENU_NORMAL_ITEM,
++  );
++  return $items;
++}
++/**
+  * Implements hook_form_node_type_form_alter().
+  *
+  * Alters node_type forms to add fields to scheduler.
+@@ -46,12 +62,15 @@ function scheduler_workbench_form_node_type_form_alter(&$form, $form_state) {
+  * Implements hook_permission().
+  */
+ function scheduler_workbench_permission() {
+-  return array(
+-    'override default scheduler time' => array(
+-      'title' => t('Override default scheduler time'),
+-      'description' => t('Allow users to override the default un-publish schedule time'),
+-    ),
++  $permissions['override default scheduler time'] = array(
++    'title' => t('Override default scheduler time'),
++    'description' => t('Allow users to override the default un-publish schedule time'),
++  );
++  $permissions['scheduler workbench'] = array(
++    'title' => t('Administer Scheduler Workbench'),
++    'description' => t('Perform administration tasks for Scheduler Workbench.'),
+   );
++  return $permissions;
+ }
+
+ /**
+@@ -252,6 +271,22 @@ function _scheduler_workbench_scheduler_unpublish(&$node) {
+  *   Node being published
+  */
+ function _scheduler_workbench_scheduler_pre_publish(&$node) {
++  // Get status allowed to be published.
++  $status = variable_get('scheduler_workbench_allowed_status', array());
++  foreach ($status as $key => $value) {
++    if ($value === 0) {
++      unset($status[$key]);
++    }
++  }
++  // Check if the revision current status allows it to be published.
++  if (!in_array($node->workbench_moderation['my_revision']->state, $status)) {
```

Comment:
The check here is fine but I think we shouldn't allow the users to define a scheduling date if the revision state is not allowed to begin with.
I would do this on scheduler_workbench_form_alter, just before calling scheduler_workbench_add_revision_publish_on_field and I would add a message to inform the user that publishing the revision in that state is not allowed.

Reply:
This allows, for example, to schedule the publication of a draft and publish it only if it is in Validation, that way you can schedule a publication and publish it only if it is in a correct status.

If a revision is not in a valid status, i will not be published and register a watchdog message with the error.

I think that we could keep the current funtionality and, as you suggest, add a message informing the user that it will no be published with the current status.